lldb/include/lldb/Core/Mangled.h

```diff
@@ -287,6 +287,18 @@ class Mangled {
   /// Retrieve \c DemangledNameInfo of the demangled name held by this object.
   const std::optional<DemangledNameInfo> &GetDemangledInfo() const;
 
+  /// Compute the base name (without namespace/class qualifiers) from the
+  /// demangled name.
+  ///
+  /// For a demangled name like "ns::MyClass<int>::templateFunc", this returns
+  /// just "templateFunc". If the demangled name is not available or the
```

Comment:
I think if you can't compute the base name you should return an empty string.  After all, if I get the base name it's because I know that I can match that accurately, so you aren't doing me any favors by returning something I can't match accurately with no way of my knowing that.